Chess.py

```diff
@@ -0,0 +1,216 @@
+        if (data != None):
+            data = data.find_parent("tr").find("td")
+            if len(data) > 1:
+                if self.participants[url]["rating"] != data[1].text:
+                    self.send_email("Rating ("+self.participants[url]["name"]+") was updated. New rating: "+data[1].text)
+                    self.participants[url]["rating"] = data[1].text
+
+        data = soup.find("td", text="Points")
+        if (data != None):
+            data = data.find_parent("tr").find("td")
+            if len(data) > 1:
+                if self.participants[url]["points"] != data[1].text:
+                    self.send_email("Points ("+self.participants[url]["name"]+") was updated. New points: "+data[1].text)
+                    self.participants[url]["points"] = data[1].text
+
+        data = soup.find("td", text="Rank")
+        if (data != None):
+            data = data.find_parent("tr").find("td")
+            if len(data) > 1:
+               if self.participants[url]["rank"] != data[1].text:
+                    self.send_email("Rank ("+self.participants[url]["name"]+") was updated. New rank: "+data[1].text)
+                    self.participants[url]["rank"] = data[1].text
+
+        data = soup.find_all("table", {"class": "CRs1"})
+        if len(data) > 2:
+            data = data[1]
+            participant_tr = data.find_all("tr")
+            if len(participant_tr) > 0:
+                participant_tr = participant_tr[-1]
+                participant = participant_tr.find_all("td")
+                if len(participant) > 0:
+                    if self.participants[url]["rd"] != participant[0].text:
+                        self.send_email("Rd. ("+self.participants[url]["name"]+") was updated. New rd.: "+data[1].text)
+                        self.participants[url]["rd"] = participant[0].text
+
+    def check_tournament (self, link, title, data):
+        for item in self.data:
+            if item["link"] == link:
+                update = False
+                if item["data"][0][1] != data[0][1]:
+                    if self.appending == False:
+                        self.send_email("Tournament \""+title+"\" was updated. New tour started: "+data[0][1])
+                    item["data"][0][1] = data[0][1]
+                    item["data"][0][0] = data[0][0]
+                    update = True
+                if item["data"][1][0] != data[1][0]:
+                    if self.appending == False:    
+                        self.send_email("Tournament \""+title+"\" was updated. New participant added: "+data[1][1])
+                    item["data"][1][0] = data[1][0]
+                    item["data"][1][1] = data[1][1]
+                    update = True
+                if update:
+                    self.data_update = True
+                return
+        self.data.append({"link": link, "title": title, "data": data})
+        self.data_update = True
+
+    def send_email (self, message):

```

Comment:
notification object should not be dependent on emails, what if you need to add more notification transports in the future?